src/main/java/data/TaskList.java

```diff
@@ -0,0 +1,116 @@
+package data;
+
+import java.util.ArrayList;
+
+/**
+ * Encapsulates TaskList that supports various methods
+ */
+
+public class TaskList {
+    private ArrayList<Task> tasks;
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Prints out the current list of tasks.
+     */
+
+    public String list() {
+        int counter = 1;
+        String result = "Here are the tasks in your list: \n";
+        for (int i = 0; i < tasks.size(); i++) {
+            if (tasks.get(i) != null) {
+                result += counter + ". " + tasks.get(i) + "\n";
+                counter++;
+            } else {
+                break;
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Delete the task with a specified index from the task list.
```

Comment:
The comments seems good and it would be better if you could add space between the description and the param description.